aten/src/ATen/cpu/vec256/vsx/vec256_double_vsx.h

```diff
@@ -0,0 +1,392 @@
+#pragma once
+
+#include <ATen/cpu/vec256/intrinsics.h>
+#include <ATen/cpu/vec256/vec256_base.h>
+#include <ATen/cpu/vec256/vsx/vsx_helpers.h>
+#include <sleef.h>
+
+namespace at {
+namespace vec256 {
+
+namespace {
+
+
+template <>
+class Vec256<double> {
+ private:
+  union {
+    struct {
+      __vd _vec0;
+      __vd _vec1;
+    };
+    struct {
+      __vllb _vecb0;
+      __vllb _vecb1;
+    };
+
+  } __attribute__((__may_alias__));
```

Comment:
Could you point to some kind of documentation that this is a valid way to use union based type aliasing? IMO this attribute like it is used here only allows the whole union to alias any other type, not that you may access the inactive member of the union. Although this might be theoretical only and may work in practice I'd still suggest to be safe and use some sort of bit_cast instead.
Besides that I'd question if that is a good way as now `Vec256<double>` is either a vector of doubles or of a bitmask. IMO that latter should be a separate type as e.g. a `log()` doesn't make sense on a bitmask.

Reply:
I do not break aliasing rules. but just in case instead of using casts (normal cast) I went with a union. As I changed this code several times it seems redundant to the whole union which, tbh, I already forgot if I was trying to make Vec casts.
As its vector SIMD, there is not any need to use bit_cast here for type punning.

Reply:
Then question: Why does/may a Vec<double> contain a "vector bool" (i.e. mask) making the union necessary? Is there any code where you treat the `_vec0` as a `_vecb0`? I only see usages of the latter in `zero_mask` and `blendv` and I don't see the need for the union there. I'm not even sure the former is correct at all but I don't know what exactly _vecb0 would contain there. 

Ok, wait, checked that. The original Vec256 requires the mask to be of the same type which is an unfortunate choice. So the attribute is not required as only valid way to use the `_vecb*` is to get it via a comparison op which creates a `vec256` with a `__vllb`-pair. So that is fine, no type punning happening in that case. Maybe a comment on the union would be helpful for others like me.